react/src/components/DateFilterComponent.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from "react";
+import { Column } from "material-table";
+import { IconButton, InputAdornment, makeStyles, TextField, Tooltip } from "@material-ui/core";
+import { NavigateBefore } from "@material-ui/icons";
+import clsx from "clsx";
+
+const useStyles = makeStyles(theme => ({
+    button: {
+        transform: "rotate(0deg)",
+        transition: theme.transitions.create("transform", {
+            duration: theme.transitions.duration.shortest,
+        }),
+    },
+    buttonFlipped: {
+        transform: "rotate(180deg)",
+    },
+}));
+
+/**
+ * A filter component for material-table for sorting before or after
+ * a specific date. Passes up a value of the format "{before,after},yyyy-mm-dd".
+ */
+export default function DateFilterComponent<RowData extends object>(props: {
+    columnDef: Column<RowData>;
+    onFilterChanged: (rowId: string, value: any) => void;
```

Comment:
The typing for `value` should be `string`, right?

Reply:
The typing here matches that of the filterComponent field of Column here: https://github.com/mbrn/material-table/blob/master/types/index.d.ts#L179. I expect it to be a string, but material-table annoyingly does it this way.

I also can't avoid typing it again by typing it as `Parameters<Column<RowData>["filterComponent"]>[0]` because it gives me a big union of other types that don't work.